--- krb5/etc/krb5.conf	2017-11-02 12:11:56.000000000 -0400
+++ krb5.conf	2020-07-05 12:16:38.479196361 -0400
@@ -19,20 +19,7 @@
 
 	pkinit_identities = ENV:PKINIT_IDENTITY
 	pkinit_identities = KEYCHAIN:
-	pkinit_identities = PKCS11:/usr/local/lib64/libcackey.so
-	pkinit_identities = PKCS11:/usr/local/lib/libcackey.so
-	pkinit_identities = PKCS11:/usr/local/lib/pkcs11/libcackey.so
-	pkinit_identities = PKCS11:/usr/lib64/pkcs11/libcackey.so
-	pkinit_identities = PKCS11:/usr/lib/pkcs11/libcackey.so
-	pkinit_identities = PKCS11:/usr/lib64/libcackey.so
-	pkinit_identities = PKCS11:/usr/lib/libcackey.so
-	pkinit_identities = PKCS11:/lib/libcackey.so
-	pkinit_identities = PKCS11:/usr/local/lib/pkcs11/libcoolkeypk11.so
-	pkinit_identities = PKCS11:/usr/lib64/pkcs11/libcoolkeypk11.so
-	pkinit_identities = PKCS11:/usr/lib/pkcs11/libcoolkeypk11.so
-	pkinit_identities = PKCS11:/usr/lib/libcoolkeypk11.so
-	pkinit_identities = PKCS11:/lib/libcoolkeypk11.so
-	pkinit_identities = PKCS11:/usr/lib/libeTPkcs11.so
+    pkinit_identities = PKCS11:/usr/lib/opensc-pkcs11.so
 
 #
 #	Certificate matching -- selecting which certificate found by identity
@@ -48,14 +35,8 @@
 #	updated to specify the location of the certificate stores (PEM format)
 #
 
-	pkinit_anchors = DIR:/usr/local/krb5/etc/CA
-	pkinit_anchors = DIR:/usr/krb5/etc/CA
-	pkinit_anchors = DIR:/usr/local/pki/ca
-	pkinit_anchors = FILE:AllDoDPKI.pem
-	pkinit_pool = DIR:/usr/local/krb5/etc/CERTS
-	pkinit_pool = DIR:/usr/krb5/etc/CERTS
-	pkinit_pool = DIR:/usr/local/pki/certs
-
+	pkinit_anchors = DIR:/usr/lib/hpcssh/krb5/etc/CA
+	pkinit_pool = DIR:/usr/lib/hpcssh/krb5/etc/CERTS
 #
 #	The rest of these entries specify the defaults per Kerberos realm
 #	for PKINIT configuration.  The values here reflect the authorization
@@ -111,7 +92,7 @@
 
 [plugins]
 	clpreauth = {
-		module = hpc_pkinit:hpcmp/hpc-pkinit-client.so
+        module = hpc_pkinit:/usr/lib/hpcssh/krb5/lib/krb5/plugins/hpcmp/hpc-pkinit-client.so
 		disable = pkinit
 	}
 #	END of Kerberos Configuration File
